examples/island/Island.cpp

```diff
@@ -273,10 +273,10 @@ float getElevation(float x, float y)
     for (int i = 0; i < perlinOctaves; i++)
     {
         elevation += stb_perlin_noise3(
-            x * perlinFrequency * std::pow(perlinFrequencyBase, i),
-            y * perlinFrequency * std::pow(perlinFrequencyBase, i),
+            x * perlinFrequency * static_cast<float>(std::pow(perlinFrequencyBase, i)),
+            y * perlinFrequency * static_cast<float>(std::pow(perlinFrequencyBase, i)),
             0, 0, 0, 0
-        ) * std::pow(perlinFrequencyBase, -i);
+        ) * static_cast<float>(std::pow(perlinFrequencyBase, -i));
```

Comment:
Why is this necessary?

`perlinFrequencyBase` is `float`, `i` is `int`.
`pow(float, int)` returns `float` (until C++11).

I see that https://en.cppreference.com/w/cpp/numeric/math/pow mentions that the return value would be promoted to `double` since C++11. How is that not a breaking change in the C++ standard library?!

Reply:
Yeah, I was confused about this as well, but we don't really want to force people to use C++03 or disable warnings, so I guess while SFML isn't using C++11, we still should handle warnings.

I guess it sort of is a breaking change?

Reply:
Ah, so we're already running with C++11 settings, OK.
Yeah then the introduced change is OK (and I could live with it causing warnings on some older compilers).